### ISSUE TITLE
Backup Data Corrupted On Restoration

### ISSUE THREAD

**Issue by p30arena** [arangodb/arangodb#2404]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [x] 3.1.16-1

On this operating system:
- [x] Windows, version: 10

These are the steps to reproduce:
1) run arangodump on an OS X machine
2) run arangorestore on a Windows 10 machine, set create database to true
3) some data gets corrupted 🔢 

Example Doc:

```
{
    "domain": "test.com",
    "uid": "b"
}
```

AQL Query:

```
for i in sites
filter i.domain == "test.com"
return i
```

NO RESULTS

but this one works:

```
for i in sites
filter i.uid == "b"
return i
```

so only domain is corrupted?!

what did I do that solved the problem?
I edited the doc and retyped test.com

**Comment by jsteemann** [comment]
How many documents are contained in the backup file for the `sites` collection?
Is it possible to provide the dump file? With that it would be much easier to tell whether the data is missing in the dump or somehow gets lost on restore, or if there is a completely different problem source.

**Comment by p30arena** [comment]
the collection contained 2 docs

https://www.dropbox.com/s/s95sfo8q9cl472a/dump%20%282%29.zip?dl=0

**Comment by jsteemann** [comment]
I have imported the dump into a 3.1 successfully. And the `sites` collection has two documents after the import. The problem is the persistent index created on the collection. The AQL query you posted makes use of it, but it seems the index is not populated on restore. It seems that this only affects the persistent indexes, not any others.

I have pushed a fix for this into the 3.1 branch a few minutes ago. The fix will be included in 3.1.17.
For now, the workaround is to drop the persistent index after the restore and re-create it manually.
